### PR TITLE
Jun.

### PR DESCRIPTION
**Pull request by juninagaki1975**
Arranged some parts.

### CODE REVIEW

**Review by kai5z — CHANGES_REQUESTED** [review]
Hi, thanks! Are you aware that these changes would be merged to the main repository?

If so: To me it looks like these would likely break the code? Also some loops that have no function? I won't merge this...

**Comment by juninagaki1975** [comment]
Thank you for your reply.
Is this mentioned about clamped center point?
> Also some loops that have no function?

File: sweep.py [hunk 1]
```diff
@@ -59,7 +59,7 @@
 def Bb(ix):
     return Matrix([[0,dNNx[ix],0],[0,0,dNNy[ix]],[0,dNNy[ix],dNNx[ix]]])
 def Bs(ix):
-    return Matrix([[dNNx[ix],-NN[ix],0],[dNNy[ix],0,-NN[ix]]])
+    return Matrix([[dNNx[ix],-dNNy[ix],0],[dNNy[ix],0,-dNNx[ix]]])
```
Comment (juninagaki1975) [inline sweep.py:62]: Components of B matrix must be derivatives all?
Replace NN with dNNx or dNNy based on some textbooks.

File: sweep.py [hunk 2]
```diff
@@ -77,8 +77,8 @@ def Bs(ix):
 expr_ke = expr_ke.applyfunc(lambda a: Jdet*integrate(integrate(a,(xi,-1,1)),(ny,-1,1)))
 
 #Element mass matrix
-for i in range(0,4):
-    for j in range(0,4):
+for j in range(0,4):
+    for i in range(0,4):
```
Comment (juninagaki1975) [inline sweep.py:81]: According to convention, swapped i and j.
(Not so affect on main work)

File: sweep.py [hunk 3]
```diff
@@ -87,8 +87,9 @@ def Bs(ix):
         expr_km[i*3+2,j*3+2] = Maa
 
 #Element force vector
-for i in range(0,4):
-    expr_f[i*3] = q*integrate(integrate(NN[i]*NN[j],(xi,-1,1)),(ny,-1,1))
+for j in range(0,4):
```
Comment (juninagaki1975) [inline sweep.py:90]: Added j loop on outer nest.
But I think it has worked without trouble when it was not modified.

File: sweep.py [hunk 4]
```diff
@@ -128,13 +129,7 @@ def Bs(ix):
 
 #Apply boundary conditions, allow sliding along a vertical axis in the middle
 ix_i = int(nDOF/3/2)*3 #The middle vertical DOF
-ix = range(ix_i+1,ix_i+3)
-K = np.delete(K, ix, axis=0)
```
Comment (juninagaki1975) [inline sweep.py:132]: Is this work like as deallocate in fortran?
I comment outed on my workspace but it must keep work on if you need these treatment...

File: sweep.py [hunk 5]
```diff
@@ -128,13 +129,7 @@ def Bs(ix):
...
-K = np.delete(K, ix, axis=1)
-M = np.delete(M, ix, axis=0)
-M = np.delete(M, ix, axis=1)
-F = np.delete(F, ix, axis=0)
-F = np.delete(F, ix, axis=1)
+ix = range(ix_i,ix_i+3)
```
Comment (juninagaki1975) [inline sweep.py:132]: The location external force applying.
When we mistake the choich of these three components in Force,
Chladni patterns may break symmtery.
(Is the order of Force components are u,v,w? It's not like...)

File: sweep.py [hunk 6]
```diff
@@ -151,12 +146,8 @@ def Bs(ix):
     MK = csr_matrix(-omega**2*M+K)
     eplo_p = spsolve(MK,F)
 
-    #Add back zeros (boundaries)
```
Comment (juninagaki1975) [inline sweep.py:154]: By addition of these clamping point, total shape of dispolacement matrix seems like disaligned.
To Kai: When we arrange and solve this disalignment, I think I can apply clamp point?
(Sorry, I am utilizing this code withou clamp point)

**Comment by kai5z** [comment]
Hi Jun. Could you please provide a picture / outtake on where this is
opened up in the textbook? Or how you derived this result using e.g.
Galerkin. Looking then at the shear force part of the stiffness matrix. To
be honest I do not directly remember how I calculated this since it was
almost 10 years ago.

File: sweep.py [hunk 7]
```diff
@@ -151,12 +146,8 @@ def Bs(ix):
...
-    eplo = np.append(eplo_p[0:ix[0]],[0] * len(ix))
-    eplo = np.append(eplo, eplo_p[ix[0]:])
-    
     #Displacements only
-    eplo = eplo[0::3]
+    eplo = eplo_p[0::3]
```
Comment (juninagaki1975) [inline sweep.py:150]: Because I removed clamped point, I employed original displacement vector directly.

File: sweep.py [hunk 8]
```diff
@@ -170,7 +161,8 @@ def Bs(ix):
     fig, ax = plt.subplots() #RdBu, bone, etc (gg python colormaps)
     Z_im = np.sqrt(np.sqrt(np.abs(Z)))
     im = plt.imshow(Z_im, cmap=cm.binary, vmin=Z_im.min(), vmax=Z_im.max(), extent=[0, w_m, 0, h_m])
-    im.set_interpolation('bicubic')#bilinear / nearest
+# Jun for debug use.
```
Comment (juninagaki1975) [inline sweep.py:164]: When debuggin it looks clear. so just use no-interpolate mode.

**Review by juninagaki1975 — COMMENTED** [review]
I added motivations on all revised points.
When you feel not suitable, please reject my requests.
But I would like you to allow me to keep utiolizing your code for my research.
In May I have Japan applied Physics meeting.
I state your name as original author but for troubles I must take in charge.

**Comment by juninagaki1975** [comment]
Because this is midnight in Japan, I would like to make answer to you tomorrow.
I am beginner for FEM so I would like to inform anything honestly.
Some part may be mixed with my own thought unclearly.

**Comment by kai5z** [comment]
Hi Jun

Please dont worry, I regrettably have very little time to use on this also
so I might not always reply quickly. I think if you can reply to the
questions I asked it should also be very useful for you.

-Kai

**Comment by juninagaki1975** [comment]
I awared my lack of courtsey, I would like to build up good communication from the begininng again.
I close this pull request and open an issue what I met on the original sweep.py.

**Pull request closed without merging**